gapis/resolve/dependencygraph2/graph_visualization/graph_visualization.go

```diff
@@ -81,6 +89,11 @@ func createGraphFromDependencyGraph(dependencyGraph dependencygraph2.DependencyG
 					newNode := getNewNode(int(nodeId), label)
 					newNode.name = name
 					newNode.attributes = attributes
+					newNode.isEndOfFrame = command.CmdFlags(ctx, api.CmdID(cmdNodeId), &api.GlobalState{}).IsEndOfFrame()
```

Comment:
Using an empty `GlobalState` actually works? 

@AWoloszyn, is this safe in general? And if so, why does `CmdFlags` have a `GlobalState` argument?

Reply:
This is not safe in general. I expect we would be ok in Vulkan, but it will likely segfault in GLES.

The way you would normally do this is get all of the EndOfFrame events from the trace (using a path.Events resolvable), and use those.
Alternatively you could insert those parameters into the dependency graph when creating it, which will have a valid global state.


`gapis/resolve/events.go`

Reply:
I will store the `CmdFlags` in the dependency graph command nodes, and switch this to get the `CmdFlags` from the dependency graph in a separate PR.